src/main.cpp

```diff
@@ -3,197 +3,101 @@
 #include "SPI_TFT_ILI9341.h"
 #include "Arial12x12.h"
 #include "supermileagelogo.h"
-#include "canID.h"
 #include <string>  
-
-DigitalOut myled(LED1);
-Serial pc(USBTX, USBRX);
-CAN can(D10, D2); 
-Ticker throttle;
-
-// !! Need to figure out interrupt priority
-InterruptIn ignition(D6);
-InterruptIn horn(D5);
-InterruptIn leftBlinker(D4);
-InterruptIn rightBlinker(D3);
-InterruptIn hazards(D9);
-InterruptIn wipers(A2);
-
-AnalogIn dms(A1); //deadman switch
-Timer debounce;
-
-
-// the TFT is connected to SPI pin 5-7
-SPI_TFT_ILI9341 TFT(D11, D12, D13, A7, D0, A3,"TFT"); 
-// mosi, miso, sclk, cs, reset, dc
-
-void setupAccessoriesInterrupts();
-void startUpSeq();
-void controlAccessories();
-void checkThrottle();
-void startCar();
-void stopCar();
-void checkDMS();
-void setupPriority();
-
-//init variables
-double FREQ = 100;
-int HIGH = 1;
-int LOW = 0;
-int IGNITION = 0;
-double dmsVAL = 0;
-int DMS = 0;
-int dmsTHRESH = 500;
-
+#include "can_common.h"
+#include "main.h"
 
 int main() {
-
     startUpSeq();
-
-    //Setup ignition interrupters
-    ignition.rise(&startCar);
-    ignition.fall(&stopCar);
-
-    //Setup accessories interrupters
-    setupAccessoriesInterrupts();
-    
-    // Set up ticker for checking throttle @ freq in Hz
-    throttle.attach(&checkThrottle, 1/FREQ);
-
-    setupPriority();
+    timer_MTR.start();
+    timer_TEL.start();
+    timer_ACC.start();
+    char prev_data_str  = ACC_task();
+    char curr_data_str  = 0;
+    CANMessage msg;
+    const char data[] = {0b11111111,0};
+    can.write(CANMessage(0x80,data,2)); 
 
     while(1){
-        // just do something random to make sure code doesn't end
-        int k = 0;
+        can.read(msg);
+        if (timer_ACC.read()> 0.2){
+            curr_data_str = ACC_task();
+            if ((curr_data_str ^ prev_data_str)){
+                const char data[] = {0, curr_data_str};
+                can.write(CANMessage(CAN_STR_ACC,data,2));
```

Comment:
I'm a little confused by which CAN ID accessories is using to receive, I thought it was 0x60 but it could also be 0xA. I will ask on Teams. I originally put this in can_common.h as CAN_ACC_OPERATION, but I think it actually makes sense to do what you did: Put messages that each module is sending under that module.